### ISSUE TITLE
ROS nano client library

### ISSUE THREAD

**Issue by vmayoral** [vmayoral/freeDDS#2]
![](http://vmayoral.github.io/nanoDDS_gitbook/img/ros_on_dds_diagrams%206.png)

We assume that a _ROS nano client library API_ should be implemented:
- How would this reduction on the library affect the ROS 2 packages?
- What are the minimal requirements for this library?

**Comment by dirk-thomas** [comment]
Regarding the diagram I would argue that nanoDDS does not cover the ROS client library part. But may be you are meaning something different with the top block?

**Comment by vmayoral** [comment]
@dirk-thomas you are right, the _ROS client nano library API_ shouldn't be part of the the DDS. The graphic should be modified.

Assuming that the _ROS client nano library API_ is out of nanoDDS, I'd say that at the very least it should provide the following:
- Nodes
- Topics
